### PR TITLE
STYLE: Fixed spelling errors in Modules/Core headers

### PR DESCRIPTION
**Pull request by dave3d**
For SimpleITK we use our SimpleITKSpellChecking package to automatically
check the comments in C++ header files for spelling errors.  Our goal is to
improve the quality of the documentation, much of which is pulled from those
header files.

You can see our spell checking code here:
https://github.com/SimpleITK/SimpleITKSpellChecking

Continuing that process, I have run the spell checker on the headers in ITK's
Modules/Core directory and fixed all the errors found.  As time goes on, I
hope to do the same on the other directories within Modules (except ThirdParty,
which has far too many bugs to deal with).

## PR Checklist
- [x] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [x] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [x] Added test (or behavior not changed)
- [x] Updated API documentation (or API not changed)
- [x] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)
- [x] Added Python wrapping to new files (if any) as described in [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) Section 9.5
- [x] Added [ITK examples](https://github.com/InsightSoftwareConsortium/ITKSphinxExamples) for all new major features (if any)

Refer to the [ITK Software Guide](https://itk.org/ItkSoftwareGuide.pdf) for
further development details if necessary.

### CODE REVIEW

File: Modules/Core/Common/include/itkCommonEnums.h [hunk 1]
```diff
@@ -219,7 +219,7 @@ class OctreeEnums
   enum class Octree : uint8_t
   {
     UNKNOWN_PLANE,   ///< The plane is Unknown
-    SAGITAL_PLANE,   ///< The plane is Sagital
+    SAGITAL_PLANE,   ///< The plane is Sagittal
```
Comment (N-Dekker) [inline Modules/Core/Common/include/itkCommonEnums.h:222]: This suggested that the identifier `SAGITAL_PLANE` might also be misspelled 🤔 Anyway, I think that's beyond the scope of this pull request!

**Review by N-Dekker — APPROVED** [review]
Thanks Dave! 👍

File: Modules/Core/Common/include/itkCommonEnums.h (hunk 1 above)
Comment (dzenanz) [reply Modules/Core/Common/include/itkCommonEnums.h:222]: Yes, we don't want to bury an interface change in this commit.

Comment (dzenanz) [reply Modules/Core/Common/include/itkCommonEnums.h:222]: I created issue #3923 to track it.

**Pull request merged**